crates/virtual-node/src/lib.rs

```diff
@@ -188,78 +185,149 @@ impl VirtualNode {
+                VirtualNode::Text(text_node) => {
+                    let current_node = element.as_ref() as &web_sys::Node;
+
+                    // We ensure that the text siblings are patched by preventing the browser from merging
+                    // neighboring text nodes. Originally inspired by some of React's work from 2016.
+                    //  -> https://reactjs.org/blog/2016/04/07/react-v15.html#major-changes
+                    //  -> https://github.com/facebook/react/pull/5753
+                    //
+                    // `ptns` = Percy text node separator
+                    if previous_node_was_text {
+                        let separator = document.create_comment("ptns");
+                        current_node
+                            .append_child(separator.as_ref() as &web_sys::Node)
+                            .unwrap();
+                    }
 
                     current_node
-                        .append_child(separator.as_ref() as &web_sys::Node)
+                        .append_child(&text_node.create_text_node())
                         .unwrap();
-                }
 
-                current_node
-                    .append_child(
-                        document
-                            .create_text_node(&child.text.as_ref().unwrap())
-                            .as_ref() as &web_sys::Node,
-                    )
-                    .unwrap();
-
-                previous_node_was_text = true;
-            } else {
-                previous_node_was_text = false;
+                    previous_node_was_text = true;
+                },
+                VirtualNode::Element(element_node) => {
+                    previous_node_was_text = false;
 
-                let child = child.create_element();
-                let child_elem = child.element;
+                    let child = element_node.create_element_node();
+                    let child_elem = child.element;
 
-                closures.extend(child.closures);
+                    closures.extend(child.closures);
 
-                element.append_child(&child_elem).unwrap();
+                    element.append_child(&child_elem).unwrap();
+                },
             }
         });
 
         CreatedElement { element, closures }
     }
 
+}
+
+impl VirtualNodeText {
     /// Return a `Text` element from a `VirtualNode`, typically right before adding it
     /// into the DOM.
     pub fn create_text_node(&self) -> Text {
         let document = web_sys::window().unwrap().document().unwrap();
-        document.create_text_node(&self.text.as_ref().unwrap())
+        document.create_text_node(&self.text)
     }
+}
 
-    /// Whether or not this `VirtualNode` is representing a `Text` node
-    pub fn is_text_node(&self) -> bool {
-        self.text.is_some()
+/// A `web_sys::Element` along with all of the closures that were created for that element's
+/// events and all of it's child element's events.
+pub struct CreatedElement {
```

Comment:
Do we still need this or can we remove it in favor of `CreatedNode` ?

Reply:
We could upcast the `Element` to a `Node` and use `CreatedNode` but I'd like to retain that type information.

But you're right about the reuse, I'll parametrize the `CreatedNode` type to avoid duplication!

Reply:
Done in aa8a00b